From 4e424f9160353b7f0f1cb00c4cbcd4b395c0ca3b Mon Sep 17 00:00:00 2001
From: Khai Wen Ng <khai.wen.ng@intel.com>
Date: Thu, 11 Sep 2025 07:05:20 +0800
Subject: [PATCH 37/51] i2c: i2c-mux: Set adapter parent to own device

Work for GMSL case. Need further checking on other use case

Signed-off-by: Khai Wen Ng <khai.wen.ng@intel.com>
---
 drivers/i2c/i2c-mux.c | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/drivers/i2c/i2c-mux.c b/drivers/i2c/i2c-mux.c
index ace64e0b0491b..4d6b2dc34af00 100644
--- a/drivers/i2c/i2c-mux.c
+++ b/drivers/i2c/i2c-mux.c
@@ -317,7 +317,7 @@ int i2c_mux_add_adapter(struct i2c_mux_core *muxc,
 	priv->adap.owner = THIS_MODULE;
 	priv->adap.algo = &priv->algo;
 	priv->adap.algo_data = priv;
-	priv->adap.dev.parent = &parent->dev;
+	priv->adap.dev.parent = muxc->dev;
 	priv->adap.retries = parent->retries;
 	priv->adap.timeout = parent->timeout;
 	priv->adap.quirks = parent->quirks;
@@ -371,13 +371,8 @@ int i2c_mux_add_adapter(struct i2c_mux_core *muxc,
 		of_node_put(mux_node);
 	}
 
-	/*
-	 * Associate the mux channel with an ACPI node.
-	 */
-	if (has_acpi_companion(muxc->dev))
-		acpi_preset_companion(&priv->adap.dev,
-				      ACPI_COMPANION(muxc->dev),
-				      chan_id);
+	if (dev_fwnode(muxc->dev))
+		device_set_node(&priv->adap.dev, dev_fwnode(muxc->dev));
 
 	if (force_nr) {
 		priv->adap.nr = force_nr;
-- 
2.51.1

